### PR TITLE
DRA pod status (KEP 4680)

### PR DESCRIPTION
**Pull request by SergeyKanzhelev**
### Description

This is a ~placeholder~ PR for KEP: https://github.com/kubernetes/enhancements/issues/4680 (alpha2)

### CODE REVIEW

**Comment by tengqm** [comment]
Doc looks good to me.
Pls provide link(s) to the upstream PR(s) so that people can help track the progress of this feature.

**Comment by hacktivist123** [comment]
Hello @pacoxu  & @tengqm 👋! I'm reaching out from the Docs team. Just checking in as we approach Docs Freeze on Tuesday November 26th 18:00 PDT. This documentation appears to still be under review. To meet the Docs Freeze, this PR must have a technical review as well as lgtm and approve labels applied, without any unaddressed comments or concerns from SIG Docs. Thank you!

**Comment by pacoxu** [comment]
/lgtm
/hold 
/cc @kannon92
to take a look as well

File: content/en/docs/concepts/scheduling-eviction/dynamic-resource-allocation.md [hunk 1]
```diff
@@ -310,6 +310,27 @@ include it.
...
+Pods that were assigned to the failed devices will continue be assigned to this device.
+It is typical that code relying on the device will start failing and Pod may get
+into Failed phase if `restartPolicy` for the Pod was not `Always` or enter the crash loop
+otherwise.
+
+By enabling the feature gate `ResourceHealthStatus`, the field `allocatedResourcesStatus`
+will be added to each container status, within the `.status` for each Pod. The `allocatedResourcesStatus`
+field reports health information for each device assigned to the container.
+
+For a failed Pod, or or where you suspect a fault, you can use this status to understand whether
```
Comment (pacoxu) [inline content/en/docs/concepts/scheduling-eviction/dynamic-resource-allocation.md:329]: `or or`

File: content/en/docs/concepts/scheduling-eviction/dynamic-resource-allocation.md [hunk 2]
```diff
@@ -310,6 +310,27 @@ include it.
...
 be installed. Please refer to the driver's documentation for details.
 
+## Devices status
+
+{{< feature-state feature_gate_name="ResourceHealthStatus" >}}
+
+There are cases when devices fail or are shut down. The responsibility of the DRA plugin
+in this case is to notify the kubelet about the situation using the `WatchResources` API.
+
+Pods that were assigned to the failed devices will continue be assigned to this device.
```
Comment (pacoxu) [inline content/en/docs/concepts/scheduling-eviction/dynamic-resource-allocation.md:320]:
```suggestion
Pods that were assigned to the failed devices will continue be assigned to these devices.
```

File: content/en/docs/concepts/scheduling-eviction/dynamic-resource-allocation.md [hunk 3]
```diff
@@ -310,6 +310,27 @@ include it.
...
+It is typical that code relying on the device will start failing and Pod may get
+into Failed phase if `restartPolicy` for the Pod was not `Always` or enter the crash loop
+otherwise.
```
Comment (shannonxtreme) [inline content/en/docs/concepts/scheduling-eviction/dynamic-resource-allocation.md:323]:
```suggestion
Code that relies on the devices will usually fail. Pods that don't set the `restartPolicy`
field to `Always` might enter the `Failed` phase. Pods that do set the `restartPolicy`
field to `Always` might enter a crash loop. 
```

Comment (shannonxtreme) [reply content/en/docs/concepts/scheduling-eviction/dynamic-resource-allocation.md:323]: Merge this into the previous paragraph to make that a single "problem statement"

File: content/en/docs/concepts/scheduling-eviction/dynamic-resource-allocation.md [hunk 4]
```diff
@@ -310,6 +310,27 @@ include it.
...
+
+By enabling the feature gate `ResourceHealthStatus`, the field `allocatedResourcesStatus`
+will be added to each container status, within the `.status` for each Pod. The `allocatedResourcesStatus`
+field reports health information for each device assigned to the container.
```
Comment (shannonxtreme) [inline content/en/docs/concepts/scheduling-eviction/dynamic-resource-allocation.md:327]:
```suggestion
If you enable the `ResourceHealthStatus` feature gate, the `allocatedResourcesStatus` field
is added to each entry in the `status.containerStatuses` field of the Pod specification. The `allocatedResourcesStatus` field reports health information for each device assigned to the
container.
```

File: content/en/docs/concepts/scheduling-eviction/dynamic-resource-allocation.md [hunk 5]
```diff
@@ -310,6 +310,27 @@ include it.
...
+
+For a failed Pod, or or where you suspect a fault, you can use this status to understand whether
+the Pod behavior may be associated with device failure. For example, if an accelerator is reporting
+an over-temperature event, the `allocatedResourcesStatus` field may be able to report this.
```
Comment (shannonxtreme) [inline content/en/docs/concepts/scheduling-eviction/dynamic-resource-allocation.md:331]:
```suggestion
For a failed Pod, or where you suspect a fault, you can use the `allocatedResourcesStatus`
field to understand whether the Pod behavior might be associated with device failure. For
example, if an accelerator is reporting an over-temperature event, the
`allocatedResourcesStatus` field might help you to identify the event.
```

**Review by shannonxtreme — COMMENTED** [review]
This is also not blocking feedback, but I think that it will help readability. If you can implement, please do, either in this PR or another. 

For docs: 

/lgtm

**Comment by shannonxtreme** [comment]
(the typos that @pacoxu pointed out might be good to fix before submission)

**Comment by reylejano** [comment]
The typos can be addressed in a follow-up PR
/approve

**Comment by haircommander** [comment]
https://github.com/kubernetes/website/pull/48861 rebases and address comments

**Comment by pohly** [comment]
/lgtm cancel

Because of https://github.com/kubernetes/website/pull/48861/files#r1859269441

**Comment by sftim** [comment]
/approve cancel

There are pending changes

@pacoxu when we hold PRs here, we prefer to state under what conditions someone could remove the hold
Could you add that detail in a comment?

**Pull request closed without merging**